npu.user.js

```diff
@@ -1288,7 +1484,61 @@
 
       /* Verify that we are indeed on a Neptun page */
       isNeptunPage: function() {
-        return document.title.indexOf("Neptun.Net") != -1;
+        return document.title.indexOf("Neptun.Net") != -1 ||
+                this.isELTEHomePage() ||
+                this.isElteLoginPage();
+      },
+
+      /* Check if we are on the new ELTE Neptun homepage. */
+      isELTEHomePage: function() {
+        const {hostname, pathname} = window.location;
+        return hostname === "neptun.elte.hu" && pathname === "/";
+      },
+
+      /* Check if we are on the new ELTE Neptun login page. */
+      isElteLoginPage: function() {
+        const {hostname, pathname} = window.location;
+        return hostname === "neptun.elte.hu" && pathname === "/Account/Login";
+      },
+
+      /*
+        Check if the user wants to log out.
+        Flag set by clicking the log out link on hallgato.neptun.elte.hu.
+      */
+      didELTERequestLogout: function() {
+        return window.location.href === "https://neptun.elte.hu/?request=logout";
+      },
+
+      /*
+        Check if the user wants to log in.
+        Flag set automatically on /Account/Login. Can be cleared by cliking the
+        'Belépés a hírekhez' button.
+      */
+      didELTERequestLogin: function() {
+        return window.location.href === "https://neptun.elte.hu/?request=login";
+      },
+
+      /* Log the user out on the new ELTE Neptun homepage. */
+      logOutELTE: function() {
+        const logoutButton = $("button:contains('Kilépés'), button:contains('Log out')");
+        logoutButton.click();
+      },
+
+      /*
+        Navigate to /Account/Login if the user is not logged in or
+        hallgato.neptun.elte.hu if the user is logged in.
+      */
+      navigateToELTELogin: function() {
+        window.location.href = "https://neptun.elte.hu/ToNeptunWeb/ToNeptunHWeb";
```

Comment:
@solymosi Please make sure that this thing here isn't hardcoded, though, or there is a way to specify which Neptun to auto log in to. Nowadays I use teacher web more often than student web and I'd not like if I'd have to manually go back either way... or if there is a way to grab if the user first opened `hallgato.neptun.elte.hu` or `oktato.neptun.elte.hu`, got redirected from there (can we catch and persist this piece of information?), and based on this, make a decision where to auto-redirect to from the script.

Reply:
No worries, I'll make the code more robust before merging and this is one of things that I'll clean up.

Reply:
The teacher Neptun old homepage is `oktato.neptun.elte.hu` and the relocating href is `/ToNeptunWeb/ToNeptun`**`O`**`Web`.